qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/FQTemplatePresenter.cpp

```diff
@@ -0,0 +1,256 @@
+  return m_model.getLocalParameters();
+}
+
+void FQTemplatePresenter::setGlobalParameters(const QStringList &globals) {
+  m_model.setGlobalParameters(globals);
+  m_view->setGlobalParametersQuiet(globals);
+}
+
+void FQTemplatePresenter::setGlobal(const QString &parName, bool on) {
+  m_model.setGlobal(parName, on);
+  m_view->setGlobalParametersQuiet(m_model.getGlobalParameters());
+}
+
+void FQTemplatePresenter::updateMultiDatasetParameters(const IFunction &fun) {
+  m_model.updateMultiDatasetParameters(fun);
+  updateViewParameters();
+}
+
+void FQTemplatePresenter::updateParameters(const IFunction &fun) {
+  m_model.updateParameters(fun);
+  updateViewParameters();
+}
+
+void FQTemplatePresenter::setCurrentDataset(int i) {
+  m_model.setCurrentDomainIndex(i);
+  updateViewParameters();
+}
+
+void FQTemplatePresenter::setDatasetNames(const QStringList &names) {
+  m_model.setDatasetNames(names);
+}
+
+void FQTemplatePresenter::setErrorsEnabled(bool enabled) {
+  m_view->setErrorsEnabled(enabled);
+}
+
+void FQTemplatePresenter::updateParameterEstimationData(
+    DataForParameterEstimationCollection &&data) {
+  m_model.updateParameterEstimationData(std::move(data));
+}
+
+void FQTemplatePresenter::updateViewParameters() {
+  if (m_model.getFitType() == "None")
+    return;
+  for (auto parameterName : m_model.getParameterNames()) {
+    m_view->setParameterValue(parameterName,
+                              m_model.getParameter(parameterName),
+                              m_model.getParameterError(parameterName));
+  }
+}
+
+QStringList FQTemplatePresenter::getDatasetNames() const {
+  return m_model.getDatasetNames();
+}
+
+double FQTemplatePresenter::getLocalParameterValue(const QString &parName,
+                                                   int i) const {
+  return m_model.getLocalParameterValue(parName, i);
+}
+
+bool FQTemplatePresenter::isLocalParameterFixed(const QString &parName,
+                                                int i) const {
+  return m_model.isLocalParameterFixed(parName, i);
+}
+
+QString FQTemplatePresenter::getLocalParameterTie(const QString &parName,
+                                                  int i) const {
+  return m_model.getLocalParameterTie(parName, i);
+}
+
+QString FQTemplatePresenter::getLocalParameterConstraint(const QString &parName,
+                                                         int i) const {
+  return m_model.getLocalParameterConstraint(parName, i);
+}
+
+void FQTemplatePresenter::setLocalParameterValue(const QString &parName, int i,
+                                                 double value) {
+  m_model.setLocalParameterValue(parName, i, value);
+}
+
+void FQTemplatePresenter::setLocalParameterTie(const QString &parName, int i,
+                                               const QString &tie) {
+  m_model.setLocalParameterTie(parName, i, tie);
+}
+
+void FQTemplatePresenter::updateView() { updateViewParameters(); }
```

Comment:
Why does this need to be a separate function

Reply:
It doesn't was just hanging around from legacy code. I've now removed updateViewParameters.